skyportal/handlers/api/gcn.py

```diff
@@ -547,15 +555,24 @@ async def get(self, dateobs=None):
             if end_date:
                 end_date = arrow.get(end_date.strip()).datetime
                 query = query.where(GcnEvent.dateobs <= end_date)
-            if tag:
+            if tag_keep:
                 tag_subquery = (
                     GcnTag.select(session.user_or_token)
-                    .where(GcnTag.text.contains(tag))
+                    .where(GcnTag.text.contains(tag_keep))
                     .subquery()
                 )
                 query = query.join(
                     tag_subquery, GcnEvent.dateobs == tag_subquery.c.dateobs
                 )
+            if tag_remove:
+                tag_subquery = (
+                    GcnTag.select(session.user_or_token)
+                    .where(GcnTag.text.contains(tag_remove))
+                    .subquery()
+                )
+                query = query.join(
+                    tag_subquery, GcnEvent.dateobs != tag_subquery.c.dateobs
+                )
```

Comment:
Could this work with just `.where(~GcnTag.text.contains(tag_remove))`?

Reply:
I did try that but it was confused!